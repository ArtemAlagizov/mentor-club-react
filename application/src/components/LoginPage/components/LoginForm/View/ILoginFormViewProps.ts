import React from "react";

export interface ILoginFormViewProps {
  email: string;
  password: string;
  onSubmitForm: React.FormEventHandler;
  onChangePassword: React.ChangeEventHandler;
  onChangeEmail: React.ChangeEventHandler;
  isAuthenticating: boolean;
  displayError: boolean;
}
