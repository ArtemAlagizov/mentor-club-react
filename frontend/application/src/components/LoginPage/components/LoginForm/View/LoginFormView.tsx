import React from "react";
import { Styled } from "../Styled";
import { ILoginFormViewProps } from "./ILoginFormViewProps";
import translator from "../../../../../config/i18next";

const LoginFormView: React.FC<ILoginFormViewProps> = ({
  onSubmitForm,
  onChangePassword,
  onChangeEmail,
  email,
  password,
  isAuthenticating,
  displayError
}): JSX.Element => {
  return (
    <Styled.LoginForm onSubmit={onSubmitForm}>
      <Styled.LoginFormInput
        type="text"
        value={email}
        onChange={onChangeEmail}
        data-id="login-email-input"
        placeholder={translator.t("loginPage.enterEmail")}
      />
      <Styled.LoginFormInput
        type="password"
        value={password}
        onChange={onChangePassword}
        data-id="login-password-input"
        placeholder={translator.t("loginPage.enterPassword")}
      />

      {displayError && (
        <Styled.LoginFormErrorMessage data-id="login-error-message">
          {translator.t("loginPage.invalidCredentials")}
        </Styled.LoginFormErrorMessage>
      )}

      <Styled.LoginFormButton
        loading={isAuthenticating}
        type="primary"
        htmlType="submit"
        size="large"
        data-id="login-submit-button"
      >
        {translator.t("loginPage.login")}
      </Styled.LoginFormButton>
    </Styled.LoginForm>
  );
};

export default LoginFormView;
