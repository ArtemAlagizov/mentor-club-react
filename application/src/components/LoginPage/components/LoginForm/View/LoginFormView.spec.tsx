import React from "react";
import { shallow } from "enzyme";
import LoginFormView from "./LoginFormView";
import { ILoginFormViewProps } from "./ILoginFormViewProps";

describe("Given LoginFormView", () => {
  describe("when rendering", () => {
    it("should have LoginFormView matching snapshot", () => {
      const props: ILoginFormViewProps = {
        onSubmitForm: jest.fn(),
        onChangePassword: jest.fn(),
        onChangeEmail: jest.fn(),
        email: "test@test.com",
        password: "fake",
        isAuthenticating: false,
        displayError: false
      };

      const component = shallow(<LoginFormView {...props} />);

      expect(component).toMatchSnapshot();
    });
  });
});
