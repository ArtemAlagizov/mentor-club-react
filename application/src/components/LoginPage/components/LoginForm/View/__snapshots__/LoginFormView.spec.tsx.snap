// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Given LoginFormView when rendering should have LoginFormView matching snapshot 1`] = `
<styled.form
  onSubmit={[MockFunction]}
>
  <styled.input
    data-id="login-email-input"
    onChange={[MockFunction]}
    placeholder="Enter email address"
    type="text"
    value="test@test.com"
  />
  <styled.input
    data-id="login-password-input"
    onChange={[MockFunction]}
    placeholder="Enter password"
    type="password"
    value="fake"
  />
  <Styled(Button)
    data-id="login-submit-button"
    htmlType="submit"
    loading={false}
    size="large"
    type="primary"
  >
    Login
  </Styled(Button)>
</styled.form>
`;
